drake/systems/frames/@CoordinateFrame/new.cpp

```diff
@@ -0,0 +1,20 @@
+#include "drakeMexUtil.h"
+#include "CoordinateFrame.h"
+
+using namespace std;
+
+void mexFunction( int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[] ) {
+  if (nrhs==1) {
+    destroyDrakeMexPointer<CoordinateFrame*>(prhs[0]);  // note: prhs[0] is the actual DrakeMexPointer object (not the CoordinateFrame object, as it would be in the non-static class methods)
```

Comment:
no.  that's unfortunately impossible.  the memory is associated with whichever mex file allocated it, and that has to be the mex file that ultimately deletes it.